packages/verkle/src/node/internalNode.ts

```diff
@@ -93,19 +105,26 @@ export class InternalNode extends BaseVerkleNode<VerkleNodeType.Internal> {
 
       const nextByteInInsertedKey = stem[this.depth + 1]
       if (nextByteInInsertedKey === nextByteInExistingKey) {
-        return newBranch.insertStem(stem, values, resolver)
+        return newBranch.insertStem(stem, values, resolver, verkleCrypto)
       }
 
       // Next word differs, so this was the last level.
       // Insert it directly into its final slot.
-      const leafNode = LeafNode.create(stem, values)
+      // TODO: Determine if this is how to create the correct commitment
+      let commitment = verkleCrypto.zeroCommitment
+      for (const [idx, value] of values.entries()) {
+        if (bytesToBigInt(value) > BIGINT_0)
+          commitment = verkleCrypto.updateCommitment(commitment, idx, new Uint8Array(32), value)
+      }
+      const leafNode = LeafNode.create(stem, values, this.depth + 1, commitment)
 
+      // TODO - Why is the leaf node set at depth + 2 instead of + 1)?
```

Comment:
I think that this is because the leaf is inserted in the position of the suffix level commitment. So we have:

depth `n` : internal node:
depth `n+1`: the extension level commitment
depth `n+2`: the suffix level commitment 

See this picture for how that corresponds to the structure:
![image](https://github.com/ethereumjs/ethereumjs-monorepo/assets/18757482/33bf11dc-c034-4b52-89f2-5ad35e17f512)